parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -1143,17 +1143,18 @@ viewRemoteBranch' (repo, sbh, path) gitBranchBehavior action = UnliftIO.try $ do
     Left err -> throwIO . C.GitSqliteCodebaseError $ C.gitErrorFromOpenCodebaseError remotePath repo err
     Right inner -> pure inner
 
--- Push a branch to a repo. Optionally attempt to set the branch as the new root, which fails if the branch is not after
+-- | Push a branch to a repo. Optionally attempt to set the branch as the new root, which fails if the branch is not after
 -- the existing root.
 pushGitBranch ::
-  forall m.
+  forall m e.
   (MonadUnliftIO m) =>
   Connection ->
-  Branch m ->
   WriteRepo ->
   PushGitBranchOpts ->
-  m (Either C.GitError ())
-pushGitBranch srcConn branch repo (PushGitBranchOpts setRoot _syncMode) = UnliftIO.try do
+  -- An action which accepts the current root branch on the remote and computes a new branch.
+  (Branch m -> m (Either e (Branch m))) ->
+  m (Either C.GitError (Either e (Branch m)))
```

Comment:
The double Either here is a bit clunky, but does the job.
I'm open to suggestions 😄

Reply:
Seems reasonable to me.